### PR TITLE
Improved Dataset classes and Windower interaction

### PR DESCRIPTION
**Pull request by gemeinl**
Created a BaseConcatDataset, which also implements a way to split into subsets.
Moved Windower out of MOABBDataset/TUHAbnoraml.
There are now functions to create windows, no more Windower classes.
See also https://github.com/braindecode/braindecode/issues/63

### CODE REVIEW

File: braindecode/datasets/base.py [hunk 1]
```diff
@@ -9,55 +9,132 @@
...
+    """Applies a windower to a base dataset.
 
     Parameters
     ----------
-    windows: ConcatDataset
-        windows/supercrops obtained throiugh application of a Windower to a
-        BaseDataset
-    info: pandas.DataFrame
+    windows: mne.Epochs
+        windows/supercrops obtained throiugh application of a windowing
```
Comment (hubertjb) [inline braindecode/datasets/base.py]:
```suggestion
        windows/supercrops obtained through application of a windowing
```

File: braindecode/datasets/base.py [hunk 2]
```diff
@@ -9,55 +9,132 @@
...
+        # split_ids are indices for WindowsDatasets
+        supercrop_ids = _windows_dataset_ids_to_supercrop_ids(
+            split_ids, self.cumulative_sizes)
+        return {split_name: Subset(self, split)
+                for split_name, split in supercrop_ids.items()}
+
+
+def _windows_dataset_ids_to_supercrop_ids(dataset_ids, cumulative_sizes):
+    supercrop_ids = {}
+    for split_name, windows_is in dataset_ids.items():
```
Comment (hubertjb) [inline braindecode/datasets/base.py]:
```suggestion
    for split_name, windows_ids in dataset_ids.items():
```

File: braindecode/datasets/base.py [hunk 3]
```diff
@@ -9,55 +9,132 @@
...
+        this_supercrop_ids = _supercrop_ids_of_windows(
+            cumulative_sizes, windows_is)
```
Comment (hubertjb) [inline braindecode/datasets/base.py]:
```suggestion
            cumulative_sizes, windows_ids)
```

File: braindecode/datasets/base.py (hunk 2 above)
Comment (hubertjb) [inline braindecode/datasets/base.py]: Not critical for now, but the `_is` prefix can be confusing - I suppose it's a pluralized version of `i`?

File: braindecode/datautil/windowers.py [hunk 4]
```diff
@@ -56,6 +32,8 @@ class EventWindower(Windower):
 
     Parameters
     ----------
+    concat_ds: BaseConcatDataset
+        a concat of base datasets each holding raw and descpription
```
Comment (hubertjb) [inline braindecode/datautil/windowers.py]:
```suggestion
        a concat of base datasets each holding raw and description
```

**Comment by gemeinl** [comment]
Closed in favor of #65.

**Pull request closed without merging**